src/memray/_memray.pyx

```diff
@@ -296,10 +296,22 @@ cdef class FileReader:
 
     cdef void _populate_allocations(self):
         cdef RecordReader* reader = self._get_reader()
-        while reader.nextRecord() not in (
-                RecordResult.RecordResultEndOfFile,
-                RecordResult.RecordResultError):
-            continue
+
+        stats = self._header["stats"]
+        self._allocations.reserve(stats["n_allocations"])
```

Comment:
I think we have changed this back and forth from the C++ to the Cython later already 3 times 😅